pymeasure/adapters/visa.py

```diff
@@ -121,18 +121,20 @@ def ask_values(self, command):
         """
         return self.connection.query_values(command)
 
-    def binary_values(self, command, header_bytes=0, dtype=np.float32):
+    def binary_values(self, command: str, header_bytes: typing.Union[str, int], dtype=np.float32):
```

Comment:
I still think that multiple arguments are clearer to support these two behaviors. We need to support the existing behavior by default.
```
    def binary_values(self, command, header_bytes=0, header_fmt=None, dtype=np.float32):
        """ Returns a numpy array from a query for binary data. By default this is implemented using the PyVISA
         `read_raw` method. If `header_fmt` is provided, the `query_binary_values` method is used instead.
 
        ...
        :param header_bytes: Integer number of bytes to ignore in header (ignored if header_format is set)
        :param header_fmt: String format of the header to parse, based on PyVISA definition (default: ignored)
        ...
        """
        if header_fmt is not None:
            return self.connection.query_binary_values(
                command,
                datatype=dtype.char,
                container=np.array,
                header_fmt=header_fmt
            )
        else:
            ...
```

Reply:
IMHO having 2 distinct mutually exclusive parameters for the same thing is semantically incorrect.

>to support the existing behavior by default.

I can set the default value to 0.